client/src/components/common/Waiting/RoomActionButtons.tsx

```diff
@@ -1,15 +1,16 @@
-import { useContext } from "react";
+import { useContext, useState } from "react";
 
 import { Auth, Room } from "../../../context";
 
-import { Col, Row, Button } from "react-bootstrap";
+import { Col, Row, Button, Spinner } from "react-bootstrap";
 import { leaveRoom as leave } from "../../../api/lobby";
 import { useNavigate } from "react-router-dom";
 import { WebSocket } from "../../../context";
 import { toast } from "react-toastify";
 import { APIResponse } from "../../../types/API";
 
 export const RoomActionButtons: React.FC = () => {
+    const [ loading, setLoading ] = useState(false);
```

Comment:
brother, use [Tanstack Query](https://tanstack.com/query/v4/docs/react/overview)

Reply:
i don't know that i only like to go brrrrr with backend scripts i'm a shitty ass in react brother
well basically i think of remaking all the front and rebuild a proper architecture for the back with the v2, since it will be necessary for the room customisation
atm i'll focus on making this shitty project also works on firefox properly brother

thanks for making me know that this thing existed tho